pkg/gatewayserver/io/ws/ws.go

```diff
@@ -119,14 +120,24 @@ func (s *srv) handleConnectionInfo(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	scheme := "ws"
+	var (
+		scheme = "ws"
+		port   = "80"
+	)
 	if r.TLS != nil || s.cfg.UseTrafficTLSAddress {
 		scheme = "wss"
+		port = "443"
+	}
+
+	// If port is retrievable from the host, use it.
+	host, p, err := net.SplitHostPort(r.Host)
```

Comment:
`host` is an unknown value if `err != nil`. You probably want to keep using `r.Host` if an error occurs (and the error means that the port is not there probably).

Reply:
Yeah well I think that my original implementation was more straightforward.

Reply:
Unfortunately it wouldn't have worked on IPv6, where `:` can be part of the host.

Reply:
Ah yeah fair.